src/Shared/TokenGenerator.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Cryptography;
+
+namespace Aspire.Hosting;
+
+internal static class TokenGenerator
+{
+    public static string GenerateToken()
+    {
+        // Generate a 128-bit entropy token 
+        var tokenBytes = GenerateEntropyToken(size: 16); // 16 bytes = 128 bits 
+
+        string tokenHex;
+#if NET9_0_OR_GREATER
+        tokenHex = Convert.ToHexStringLower(tokenBytes); 
+#else
+        tokenHex = Convert.ToHexString(tokenBytes).ToLowerInvariant();
+#endif
+
+        return tokenHex;
+    }
+
+    private static byte[] GenerateEntropyToken(int size)
+    {
+#if NET6_0_OR_GREATER
+        return RandomNumberGenerator.GetBytes(size);
+#else
+        using (var rng = new RNGCryptoServiceProvider()) 
+        { 
+            byte[] token = new byte[size]; 
+            rng.GetBytes(token); 
+            return token; 
+        } 
+#endif
```

Comment:
We aren't building anything on < net6.0 in this repo.

```suggestion
        return RandomNumberGenerator.GetBytes(size);
```

Reply:
Client load balancing has netstandard2.0.

I'd like to leave it as-is so people who want to have the same token generator can copy this code and use it on older frameworks, e.g. .NET Framework.

Reply:
> Client load balancing has netstandard2.0.

Do you have a link? I don't see that in this repo.

This is dead code, and I don't see the value in keeping it. Especially when we go to .NET 9 and we can opitmize this code to only 1 alloc by stackalloc'ing the buffer, putting random bytes into it, and then the single call to `ToHexStringLower(ReadOnlySpan<byte> bytes)` passing in the buffer.

Reply:
> Do you have a link? I don't see that in this repo.

I was wrong. I think what I remembered was someone asking for netstandard2.0 support.